### PR TITLE
Readme: adds documentation for Board, parameters and options. Closes gh-159

### PR DESCRIPTION
**Pull request by rwaldron**
Signed-off-by: Rick Waldron <waldron.rick@gmail.com>

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -23,34 +23,124 @@ For global cli use:
...
+var board = new Board(new Serialport(...));
 
-board.on("ready", function() {
+board.on("ready", () => {
   // Arduino is ready to communicate
 });
 ```
 
-# REPL
+#### With an Etherport object:
```
Comment (soundanalogous) [inline readme.md]: May be useful to add that Etherport is a TCP server that can be used with StandardFirmataEthernet or StandardFirmataWiFi when configured as a TCP client (StandardFirmataEthernet can currently only be configured as a TCP client). There is also [etherport-client](https://github.com/mwittig/etherport-client) which is a TCP client that can be used with StandardFirmataWiFi when configured as a TCP server.

Comment (rwaldron) [inline readme.md]: Sure thing.

Comment (rwaldron) [inline readme.md]: I'm going to copy and paste most of this ;)

**Comment by rwaldron** [comment]
@soundanalogous I added your notes about Etherport and Etherport-client

**Comment by soundanalogous** [comment]
LGTM

**Comment by soundanalogous** [comment]
@rwaldron are there any other changes to be made here or is this ready to merge?

**Pull request merged**